hotsos/core/host_helpers/systemd.py

```diff
@@ -86,25 +86,40 @@ def start_time_secs(self):
         return t.timestamp()
 
     @property
-    def memory_current(self):
-        """ Returns service current memory usage in bytes. """
-        path = os.path.join(HotSOSConfig.data_root,
-                            'sys/fs/cgroup/memory/system.slice',
-                            "{}.service".format(self.name),
-                            'memory.usage_in_bytes')
+    def memory_current_kb(self):
+        """ Returns service current memory usage in kbytes.
+
+        See https://www.kernel.org/doc/Documentation/cgroup-v1/memory.txt
+        """
+        cgroup_base = os.path.join(HotSOSConfig.data_root, 'sys/fs/cgroup')
+        path = os.path.join(cgroup_base,
+                            "memory/system.slice/{}.service".format(self.name),
+                            'memory.stat')
         if not os.path.exists(path):
             # path changed between Ubuntu Focal and Jammy releases.
-            path = os.path.join(HotSOSConfig.data_root,
-                                'sys/fs/cgroup/system.slice',
+            path = os.path.join(cgroup_base, 'system.slice',
                                 "{}.service".format(self.name),
-                                'memory.current')
+                                'memory.stat')
 
-        if not os.path.exists(path):
-            log.warning("service memory info not found at %s", path)
-            return 0
+        total_usage = {}
+        fs = FileSearcher()
+        fs.add(SearchDef(r'(cache|rss|swap) (\d+)'), path=path)
+        for result in fs.run().get(path, {}):
+            total_usage[result.get(1)] = int(result.get(2))
+
+        if len(total_usage) == 0:
+            log.warning("failed to identify mem usage info for %s in %s",
+                        "{}.service".format(self.name), path)
+
+        if len(total_usage) < 3:
```

Comment:
swap counter will only exist if swap is enabled so we should not print this warning